src/manifest.rs

```diff
@@ -33,15 +33,212 @@ use crate::{
 /// The name of the manifest file
 pub const MANIFEST_FILE: &str = "Proto.toml";
 
-/// A `buffrs` manifest format used for serialization and deserialization.
+/// The canary edition supported by this version of buffrs
+pub const CANARY_EDITION: &str = concat!("0.", env!("CARGO_PKG_VERSION_MINOR"));
+
+/// Edition of the buffrs manifest
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Edition {
+    /// The canary edition of manifests
+    ///
+    /// This indicates that breaking changes and unstable behavior can occur
+    /// at any time. Users are responsible for consulting documentation and
+    /// help channels if errors occur.
+    Canary,
+    /// Unknown edition of manifests
+    ///
+    /// This is unrecommended as breaking changes could be introduced due to being
+    /// in the beta release channel
+    Unknown,
+}
+
+impl Default for Edition {
```

Comment:
Nit: I would maybe avoid this to enforce concious choice of edition by callers. What about `Edition::latest()`